### PR TITLE
perf: move sentry/monitors to orjson

### PR DESCRIPTION
**Pull request by anonrig**
Splitting the changes from https://github.com/getsentry/sentry/pull/71185 into multiple pull-requests as a recommendation from @fpacifici (ref: https://github.com/getsentry/sentry/pull/71185#issuecomment-2125854962).

This pull-request only changes `json` usages in `src/sentry/monitors` folder.

### CODE REVIEW

**Review by wedamija — APPROVED** [review]
Looks good to me. I'd recommend not merging this Friday just in case we have issues, so either today or Monday

File: src/sentry/monitors/processing_errors/manager.py [hunk 1]
```diff
@@ -98,7 +99,9 @@ def store_error(error: CheckinProcessingError, monitor: Monitor | None):
     entity_identifier = _get_entity_identifier_from_error(error, monitor)
     error_set_key = build_set_identifier(entity_identifier)
     error_key = build_error_identifier(error.id)
-    serialized_error = json.dumps(error.to_dict())
+    serialized_error = orjson.dumps(
+        error.to_dict(), option=orjson.OPT_UTC_Z | orjson.OPT_NON_STR_KEYS
```
Comment (wedamija) [inline src/sentry/monitors/processing_errors/manager.py:103]: I'm not sure if there are efficiency improvements here, but I think it's safe to not have the `OPT_NON_STR_KEYS` option for this code.

Comment (anonrig) [reply src/sentry/monitors/processing_errors/manager.py]: I've removed it. I mostly added it as a `defensive` approach.

Comment (anonrig) [reply src/sentry/monitors/processing_errors/manager.py]: Thanks for the pull-request review @wedamija!

**Comment by anonrig** [comment]
cc @wedamija this requires an additional review, since another `json.dumps` has been landed before I get to land this pull-request.

**Review by wedamija — APPROVED** [review]

**Pull request merged**